Adds extra MFP macros for MFP configuration that should be done on the Chumby
diff -rup linux-3.13-rc1/arch/arm/mach-mmp/include/mach/mfp-pxa168.h linux-3.13-rc1/arch/arm/mach-mmp/include/mach/mfp-pxa168.h
--- linux-3.13-rc1/arch/arm/mach-mmp/include/mach/mfp-pxa168.h	2013-11-22 11:30:55.000000000 -0800
+++ linux-3.13-rc1/arch/arm/mach-mmp/include/mach/mfp-pxa168.h	2013-11-29 22:38:42.314144325 -0800
@@ -17,6 +17,9 @@
 #define MFP_CFG_DRV(pin, af, drv)	\
 	(MFP_LPM_INPUT | MFP_PIN(MFP_PIN_##pin) | MFP_##af | MFP_DRIVE_##drv)
 
+#define MFP_CFG_DRV_PULL(pin, af, drv, pull)	\
+	(MFP_LPM_FLOAT | MFP_PIN(MFP_PIN_##pin) | MFP_##af | MFP_DRIVE_##drv | MFP_LPM_PULL_##pull | MFP_PULL_##pull)
+
 /* GPIO */
 #define GPIO0_GPIO		MFP_CFG(GPIO0, AF5)
 #define GPIO1_GPIO		MFP_CFG(GPIO1, AF5)
@@ -244,6 +247,29 @@
 #define	GPIO32_MMC2_DAT2	MFP_CFG_DRV(GPIO32, AF6, FAST)
 #define	GPIO33_MMC2_DAT3	MFP_CFG_DRV(GPIO33, AF6, FAST)
 
+/* MMC2 */
+#define	GPIO90_MMC2_DAT3	MFP_CFG_DRV(GPIO90, AF1, FAST)
+#define	GPIO91_MMC2_DAT2	MFP_CFG_DRV(GPIO91, AF1, FAST)
+#define	GPIO92_MMC2_DAT1	MFP_CFG_DRV(GPIO92, AF1, FAST)
+#define	GPIO93_MMC2_DAT0	MFP_CFG_DRV(GPIO93, AF1, FAST)
+#define	GPIO94_MMC2_CMD		MFP_CFG_DRV(GPIO94, AF1, FAST)
+#define	GPIO95_MMC2_CLK		MFP_CFG_DRV(GPIO95, AF1, FAST)
+
+/* MMC3 */
+#define GPIO0_MMC3_DAT7		MFP_CFG_DRV_PULL(GPIO0, AF6, FAST, HIGH)
+#define GPIO1_MMC3_DAT6		MFP_CFG_DRV_PULL(GPIO1, AF6, FAST, HIGH)
+#define GPIO2_MMC3_DAT5		MFP_CFG_DRV_PULL(GPIO2, AF6, FAST, HIGH)
+#define GPIO3_MMC3_DAT4		MFP_CFG_DRV_PULL(GPIO3, AF6, FAST, HIGH)
+#define GPIO4_MMC3_DAT3		MFP_CFG_DRV_PULL(GPIO4, AF6, FAST, HIGH)
+#define GPIO5_MMC3_DAT2		MFP_CFG_DRV_PULL(GPIO5, AF6, FAST, HIGH)
+#define GPIO6_MMC3_DAT1		MFP_CFG_DRV_PULL(GPIO6, AF6, FAST, HIGH)
+#define GPIO7_MMC3_DAT0		MFP_CFG_DRV_PULL(GPIO7, AF6, FAST, HIGH)
+#define GPIO8_MMC3_CLK		MFP_CFG_DRV(GPIO8, AF6, FAST)
+#define GPIO9_MMC3_CMD		MFP_CFG_DRV_PULL(GPIO9, AF6, FAST, HIGH)
+
+/* ? */
+#define GPIO16_SMC_nCS0_DIS	MFP_CFG_DRV_PULL(GPIO16, AF0, FAST, HIGH)
+
 /* MMC4 */
 #define GPIO125_MMC4_DAT3       MFP_CFG_DRV(GPIO125, AF7, FAST)
 #define GPIO126_MMC4_DAT2       MFP_CFG_DRV(GPIO126, AF7, FAST)
